addons/web/static/src/legacy/scss/bootstrap_review.scss

```diff
@@ -208,19 +208,37 @@ $o-navbar-nav-pills-link-border-radius: $nav-pills-border-radius !default;
     // body is white, the text will not be visible on modals with a white
     // background.
 
-    // Check if the body text color is light.
-    $-is-body-text-light: color-yiq($body-color) == $yiq-text-dark;
-
     @if $modal-content-color == null {
-        color: if(
-            $-is-body-text-light,
-            color-yiq($modal-content-bg, $light: $body-color, $dark: $yiq-text-dark),
-            color-yiq($modal-content-bg, $light: $yiq-text-light, $dark: $body-color)
-        );
-    }
-    @if $-is-body-text-light {
-        .text-muted {
-            color: color-yiq($modal-content-bg, $light: $text-muted, $dark: mute-color($yiq-text-dark)) !important;
+        color: adjust-body-text-color-for-background($modal-content-bg);
+    }
+
+    .text-muted {
+        color: adjust-muted-text-color-for-background($modal-content-bg) !important;
+    }
+}
+
+// Popover
+.popover {
+    // The popover can have a different background color than that of the body.
+    // Here, we adjust the text color of the popover in case the body color
+    // (used by default for the text color of popovers) is not visible inside a
+    // popover due to a lack of contrast (e.g. on a website with a dark
+    // background).
+    @if $popover-header-color == null {
+        .popover-header {
+            color: adjust-body-text-color-for-background($popover-header-bg);
+        }
+    }
+    @if $popover-body-color == $body-color {
+        .popover-body {
+            color: adjust-body-text-color-for-background($popover-bg);
```

Comment:
While you are at it, should not should fix text-muted inside popover bodies too? Would seem consistent with the rest, no?

Reply:
> While you are at it, should not should fix text-muted inside popover bodies too? Would seem consistent with the rest, no?

I preferred to only address known issues in the stable version to minimize any potential impact on the custom style implemented by clients. However, I'll let you decide.